packages/@lwc/engine-dom/src/apis/create-element.ts

```diff
@@ -63,6 +64,9 @@ function monkeyPatchDomAPIs() {
             return callNodeSlot(appendedNode, ConnectingSlot);
         },
         insertBefore(newChild, referenceNode) {
+            if (arguments.length < 2) {
```

Comment:
```suggestion
            if (process.env.NODE_ENV !== 'production' && arguments.length < 2) {
```

We don't need this warning in production mode.